### ISSUE TITLE
bug: slot error not working properly in ion-item in some cases

### ISSUE THREAD

**Issue by andreas-aeschlimann** [ionic-team/ionic-framework#24445]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

The new slot "error" does not work when using an `ion-input` in the same `ion-icon`. The slot "helper" seems to work normally.

```
<ion-item>
     <ion-label [position]="'fixed'">Label</ion-label>
     <ion-input></ion-input>
     <ion-note slot="error">Test</ion-note
</ion-item>
```

In this example, the "Test" text is not visible at all. When I use an `ion-label` instead of an `ion-note` for the error, it shows the error element but it only has 50% width of the full `ion-item`.

### Expected Behavior

Items with the error slot should show up as expected.

### Steps to Reproduce

See above.

### Ionic Info

Ionic:

   Ionic CLI                     : 6.18.1 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.1
   @angular-devkit/build-angular : 12.2.12
   @angular-devkit/schematics    : 12.2.12
   @angular/cli                  : 12.2.12
   @ionic/angular-toolkit        : 5.0.3

Capacitor:

   Capacitor CLI      : 3.3.3
   @capacitor/android : not installed
   @capacitor/core    : 3.3.3
   @capacitor/ios     : not installed

Utility:

   cordova-res (update available: 0.15.4) : 0.15.1
   native-run                             : 1.5.0

System:

   NodeJS : v17.0.1 (/usr/local/Cellar/node/17.0.1/bin/node)
   npm    : 8.1.0
   OS     : macOS Monterey

**Comment by sean-perkins** [comment]
Hello @andreas-aeschlimann, thanks for the issue!

This may just be poor documentation on our part.

The `error` slot with `ion-item` is intended to be used with a form and currently our only direct integration is with Angular Forms. When using form controls, Ionic will map the `ng-invalid` class to `ion-invalid` on the item. This class is the missing piece to make the `error` slot visible.

If you update your example to this:
```html
<ion-item class="ion-invalid">
  <ion-label position="fixed">Label</ion-label>
  <ion-input></ion-input>
  <ion-note slot="error">Test</ion-note>
</ion-item>
```

You will see the error slot always.

Let me know if this resolves the issue for you. I don't believe this is a bug, as we don't want to display the error contents if the form control isn't in an error state.

**Comment by andreas-aeschlimann** [comment]
Ah yeah, I guess I could not know this. So it does not look like a code bug, but a documentation problem.

It is important to note that the error text also shows when the form control has not been touched yet. This is a bit counter-intuitive, because the form control borders will not turn red until you touch it (or click).

**Comment by sean-perkins** [comment]
@andreas-aeschlimann I'll need to consult with the team on this portion.

If we change the behavior of how the error slot is displayed to require touch, that would require existing implementations that show all errors upfront to manually call `markAllAsTouched` for the form group. This would also be another level of detail for non-Angular implementations to maintain.

With the existing implementation you can hide the error until touch, but that also requires more to implement:
```html
<ion-item>
  <ion-input formControlName="controlName"></ion-input>
  <ion-note *ngIf="form.get('controlName').touched" slot="error">Error message</ion-note>
</ion-item>
```

**Comment by vildhjarta8** [comment]
It still not working even inside the form group on the real iOS device or simulator. 
`ion-item` contains `ion-invalid` class but `ion-note[slot="error"]` still not visible at all. It behaves as expected in the browser only.
@ionic/angular 6.0.3, Angular 13

**Comment by andreas-aeschlimann** [comment]
> It still not working even inside the form group on the real iOS device or simulator. `ion-item` contains `ion-invalid` class but `ion-note[slot="error"]` still not visible at all. It behaves as expected in the browser only. @ionic/angular 6.0.3, Angular 13

Are you sure? I noticed it works only if there is a form component (for example ion-input) present.

**Comment by sean-perkins** [comment]
@vildhjarta8 I'm seeing the same issue as you. It's a Safari issue with style priority (can see with web browser). 

I'll look into the fix. I may need to do some repurposing/splitting of this issue, because we have a few different conversations around the `slot="error"` happening and I don't want to miss out on those details.

**Comment by sean-perkins** [comment]
Hello everyone 👋  follow-up to close out this issue.

1. I've created a new issue in Ionic Docs to track the lack of documentation surrounding adding `.ion-invalid` to the `ion-item` to make the error slot visible: https://github.com/ionic-team/ionic-docs/issues/2186
2. The other mentioned issue regarding the slot not being visible on device was resolved in this PR: #24579

Appreciate everyone's involvement here with discussing and diagnosing the issue at hand!
